manager/logbroker/broker.go

```diff
@@ -383,6 +383,15 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) (err er
 			}
 		}
 
+		// if we have a close message, close out the subscription
+		if log.Close {
+			// Mark done and then set to nil so if we error after this point,
```

Comment:
Can't comment on the code above because of GH, but anyway.

In the defer we have:
```
 			currentSubscription.Done(remote.NodeID, err)
```

Shouldn't we **always** set an error? If we get to the defer, it means the agent unexpectedly dropped out.